src/test/unit/BLSPubkeyRegistryUnit.t.sol

```diff
@@ -0,0 +1,211 @@
+    }
+
+    function testCallRegisterOperatorFromNonCoordinatorAddress(address nonCoordinatorAddress) public {
+        cheats.assume(nonCoordinatorAddress != address(registryCoordinator));
+
+        cheats.startPrank(nonCoordinatorAddress);
+        cheats.expectRevert(bytes("BLSPubkeyRegistry.onlyRegistryCoordinator: caller is not the registry coordinator"));
+        blsPubkeyRegistry.registerOperator(nonCoordinatorAddress, new uint8[](0), BN254.G1Point(0, 0));
+        cheats.stopPrank();
+    }
+
+    function testCallDeregisterOperatorFromNonCoordinatorAddress(address nonCoordinatorAddress) public {
+        cheats.assume(nonCoordinatorAddress != address(registryCoordinator));
+
+        cheats.startPrank(nonCoordinatorAddress);
+        cheats.expectRevert(bytes("BLSPubkeyRegistry.onlyRegistryCoordinator: caller is not the registry coordinator"));
+        blsPubkeyRegistry.deregisterOperator(nonCoordinatorAddress, new uint8[](0), BN254.G1Point(0, 0));
+        cheats.stopPrank();
+    }
+
+    function testOperatorDoesNotOwnPubKeyRegister(address operator) public {
+        cheats.startPrank(address(registryCoordinator));
+        cheats.expectRevert(bytes("BLSRegistry._registerOperator: operator does not own pubkey"));
+        blsPubkeyRegistry.registerOperator(operator, new uint8[](1), BN254.G1Point(1, 0));
+        cheats.stopPrank();
+    }
+    function testOperatorDoesNotOwnPubKeyDeregister(address operator) public {
+        cheats.startPrank(address(registryCoordinator));
+        cheats.expectRevert(bytes("BLSRegistry._deregisterOperator: operator does not own pubkey"));
+        blsPubkeyRegistry.deregisterOperator(operator, new uint8[](1), BN254.G1Point(1, 0));
+        cheats.stopPrank();
+    }
+
+    function testOperatorRegisterZeroPubkey() public {
+        cheats.startPrank(address(registryCoordinator));
+        cheats.expectRevert(bytes("BLSRegistry._registerOperator: cannot register zero pubkey"));
+        blsPubkeyRegistry.registerOperator(defaultOperator, new uint8[](1), BN254.G1Point(0, 0));
+        cheats.stopPrank();
+    }
+    function testRegisteringWithNoQuorums() public {
+        cheats.startPrank(address(registryCoordinator));
+        cheats.expectRevert(bytes("BLSRegistry._registerOperator: must register for at least one quorum"));
+        blsPubkeyRegistry.registerOperator(defaultOperator, new uint8[](0), BN254.G1Point(1, 0));
+        cheats.stopPrank();
+    }
+
+    function testDeregisteringWithNoQuorums() public {
+        cheats.startPrank(address(registryCoordinator));
+        cheats.expectRevert(bytes("BLSRegistry._deregisterOperator: must register for at least one quorum"));
+        blsPubkeyRegistry.deregisterOperator(defaultOperator, new uint8[](0), BN254.G1Point(1, 0));
+        cheats.stopPrank();
+    }
+
+    function testRegisterOperatorBLSPubkey(address operator) public {
+        bytes32 pkHash = BN254.hashG1Point(defaultPubKey);
```

Comment:
how feel safer if we had a random uint256 input which we hashed to a g1 pubkey as the public key

Reply:
do you mean fuzz the actual G1Point, as in x and y?

Reply:
yep